### PR TITLE
Add support for LESS variables to be defined by aggregator config.

### PR DESCRIPTION
**Pull request by chuckdumont**
Supports the `lessGlobals` property in the client-side AMD loader config (used when compiling LESS files on the client as when bypassing the aggregator) and the server-side AMD config.  `lessGlobals` may specify a property map of LESS variable name-value pairs (both the client and server configs), or a function (server config only).  If it specifies a function, then the function returns the property map of LESS variable name-value pairs.  The javascript `has()` function is defined within the scope of the function and may be called to conditionally define properties based on the features specified by the request.  For example:
```
lessGlobals: function() {
   return {
      bidiLeft: has('RtlLanguage') ? 'right' : 'left'
   };
}
```

### CODE REVIEW

**Comment by chuckdumont** [comment]
@ddumont, @knpwrs  please review.

**Comment by chuckdumont** [comment]
I updated this pull request to include support for variable names in @import statements.  This required modifying the LESS module builder to provide a file loader to the LESS compiler and letting the LESS compiler import LESS files rather than inlining the imports before invoking the LESS compiler like we were doing.

**Comment by chuckdumont** [comment]
Note that specifying a global variable that will be used in an @import name is a bit tricky if the name can contain the '/' character.  That's because LESS treats '/' as a division operator.  So you have to double quote it when specifying the value in a global variable defined in the aggregator config.  For example:
```
lessGlobals: {
   myImportPath: '\"foo/bar\"'
}
```
Then you can use the `myImportName` variable in an @import statement as follows:
```
@import: '@{myImportPath}/mixin.less';
```

File: jaggr-core/pom.xml [hunk 1]
```diff
@@ -303,7 +303,7 @@
               <directory>.</directory>
               <includes>
                 <include>lib/**</include>
-                <include>src/main/resources/less*.js</include>
+                <include>src/main/resources/less-rhino-*.js</include>
```
Comment (ddumontatibm) [inline jaggr-core/pom.xml]: what happened here?

Comment (chuckdumont) [inline jaggr-core/pom.xml]: The directory now contains the file lessFileLoader.js so I had to make the delete selector more specific.

**Comment by ddumontatibm** [comment]
lgtm

**Pull request merged**